### PR TITLE
Update the ExternalSecret status even when data is empty

### PR DESCRIPTION
**Pull request by shuheiktgw**
## Problem Statement

The details are described in the issue below, but in short, I believe we should update the ES status when the dataMap is empty even if deletion policy is retain. Please check [my comment](https://github.com/external-secrets/external-secrets/issues/2874#issuecomment-1836925600) as well 🙇 

## Related Issue

Closes https://github.com/external-secrets/external-secrets/issues/2874

## Proposed Changes

Update the status properly.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: pkg/controllers/externalsecret/externalsecret_controller.go [hunk 1]
```diff
@@ -230,21 +230,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 				return ctrl.Result{}, err
 			}
 
-			err = r.Delete(ctx, secret)
-			if err != nil && !apierrors.IsNotFound(err) {
+			if err := r.Delete(ctx, secret); err != nil && !apierrors.IsNotFound(err) {
 				r.markAsFailed(log, errDeleteSecret, err, &externalSecret, syncCallsError.With(resourceLabels))
+				return ctrl.Result{}, err
```
Comment (shuheiktgw) [inline pkg/controllers/externalsecret/externalsecret_controller.go:235]: This is not directly related to this PR, but I believe we should return the error instead of ignoring it. Otherwise, the status would be ready even if the operator failed to delete the secret.

**Comment by gusfcarvalho** [comment]
This problem was a bit invisible because our validating webhook for `ExternalSecrets` blocks empty data/dataFrom in the first place: https://github.com/external-secrets/external-secrets/blob/main/apis/externalsecrets/v1beta1/externalsecret_validator.go#L55-L57 - we should update this one as well.

another thing I'd like to know is what happens with ExternalSecrets with (fixed values) templates but no data - should that render a Secret Manifest with the fixed value? (IIRC this was what started the issue in the first place)

**Comment by moolen** [comment]
> another thing I'd like to know is what happens with ExternalSecrets with (fixed values) templates but no data - should that render a Secret Manifest with the fixed value? (IIRC this was what started the issue in the first place)

Well, ESO is not a template engine (or is it? :smile:). I guess it would need some more rework to make it a template engine, though i would expect that it should work: defining just the `target.template.data`
